### DISCUSSION TITLE
Test: `assertViewHas` does not work properly

### DISCUSSION THREAD

**Discussion by robsontenorio** [livewire/livewire#6854] [7. Bugs (archived)] [answered]
### Livewire version

3.0.5

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

The example from docs for `assertViewHas` does not works properly for Volt.

https://livewire.laravel.com/docs/testing#asserting-data-from-the-view

```php
Volt::test('somecomponent')
      ->assertViewHas('posts', function ($posts) {
          $this->assertEquals(2, 2);
      });

// ------> Also fails for `Livewire::test('somecomponent')`
```

It fails.

```
 Failed asserting that null is true.
```

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@robsontenorio Please open this on the volt repo, as we don't manage volt, the Laravel teams does 🙂

**Reply by robsontenorio** [reply]
@joshhanley  Hi Josh!

This problem also happens with `Livewire::test(...)->`

**Reply by jaceju** [reply, marked as answer]
@robsontenorio I solved this problem with `return true;` in the closure:

```php
Volt::test('somecomponent')
      ->assertViewHas('posts', function ($posts) {
          $this->assertEquals(2, 2);
          return true;
      });
```

Because the `assertViewHas` method uses `PHPUnit::assertTrue` to assert the return result of the closure, so you must use `return true;` to avoid this error.

FYI.
